### DISCUSSION TITLE
Get the CorrelationId of successful requests

### DISCUSSION THREAD

**Discussion by StefanSchoof** [pnp/pnpcore#1460] [Q&A] [answered]
For Audit purpose I like to save the SharePoint CorrelationId of successful request with some additional data from my application. 

Is there an way to get or set the SharePoint CorrelationId in pnpcore for successful requests?

**Comment by jansenbe** [comment, marked as answer]
@StefanSchoof : thanks for your PR. I did include a generic method that brings selected response headers for all API calls to the SDK user, this can also be useful for you (unless you only need the correlationid for batch requests). See https://pnp.github.io/pnpcore/using-the-sdk/basics-responseheaders.html

**Reply by StefanSchoof** [reply]
Looks great. Thanks.
